docs/topics/compatibility-guides/compatibility-guide-22.md

```diff
@@ -0,0 +1,574 @@
+
+### Deprecate `publishAllLibraryVariants()` function
+
+> **Issue**: [KT-60623](https://youtrack.jetbrains.com/issue/KT-60623)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The `publishAllLibraryVariants()` function is deprecated. It was designed to publish all build variants
+> for Android targets. This approach is not recommended now, as it can cause problems with variant resolution, especially
+> when multiple flavors and build types are used. Use the `publishLibraryVariants()` function, which specifies build variants,
+> instead.
+>
+> **Deprecation cycle**:
+>
+> - 2.2.0: `publishAllLibraryVariants()` is deprecated
+
+### Deprecate `android` target
+
+> **Issue**: [KT-71608](https://youtrack.jetbrains.com/issue/KT-71608)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The `android` target name is deprecated in the current Kotlin DSL. Use `androidTarget` instead.
+>
+> **Deprecation cycle**:
+>
+> - 1.9.0: introduce a deprecation warning when the `android` name is used in Kotlin Multiplatform projects 
+> - 2.1.0: raise this warning to an error
+> - 2.2.0: remove the `android` target DSL from the Kotlin Multiplatform Gradle plugin
+
+### Deprecate `konanVersion` in `CInteropProcess`
+
+> **Issue**: [KT-71069](https://youtrack.jetbrains.com/issue/KT-71069)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The `konanVersion` property in the `CInteropProcess` task is deprecated.
+> Use `CInteropProcess.kotlinNativeVersion` instead.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: report a warning when `konanVersion` is used
+> - 2.2.0: raise this warning to an error
+> - 2.3.0: remove `konanVersion` property from the Kotlin Gradle plugin
+
+### Deprecate `destinationDir` in `CInteropProcess`
+
+> **Issue**: [KT-71068](https://youtrack.jetbrains.com/issue/KT-71068)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The `destinationDir` property in the `CInteropProcess` task is deprecated.
+> Use `CInteropProcess.destinationDirectory.set()` instead.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: report a warning when `destinationDir` is used
+> - 2.2.0: raise this warning to an error
+> - 2.3.0: remove `destinationDir` from the Kotlin Gradle plugin
+
+### Deprecate `kotlinArtifacts` API
+
+> **Issue**: [KT-74953](https://youtrack.jetbrains.com/issue/KT-74953)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The experimental `kotlinArtifacts` API is deprecated. Use the current DSL available in the Kotlin Gradle
+> plugin to [build final native binaries](https://www.jetbrains.com/help/kotlin-multiplatform-dev/multiplatform-build-native-binaries.html).
+> If it's not sufficient for migration, leave a comment in [this YT issue](https://youtrack.jetbrains.com/issue/KT-74953).
```

Comment:
@danil-pavlov by "it" do you mean the current DSL?

Reply:
yes, correct